This PR doesn't pass the tests on my machine

```
=========================================================================================================================================================================== test session starts ============================================================================================================================================================================
platform darwin -- Python 3.10.13, pytest-7.2.1, pluggy-1.0.0 -- /Users/harrylees/Documents/GitHub/pre-commit-pin-dependencies/.venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/harrylees/Documents/GitHub/pre-commit-pin-dependencies
collected 9 items

tests/iter_dependencies_test.py::test_iter_dependencies FAILED                                                                                                                                                                                                                                                                                                       [ 11%]
tests/iter_dependencies_test.py::test_iter_dependencies_no_dependencies PASSED                                                                                                                                                                                                                                                                                       [ 22%]
tests/iter_dependencies_test.py::test_iter_dependencies_unexpected_version_type PASSED                                                                                                                                                                                                                                                                               [ 33%]
tests/pin_dependency_test.py::test_no_dependencies PASSED                                                                                                                                                                                                                                                                                                            [ 44%]
tests/pin_dependency_test.py::test_pinned_dependencies PASSED                                                                                                                                                                                                                                                                                                        [ 55%]
tests/pin_dependency_test.py::test_unpinned_dependencies[tests/resources/caret.toml] PASSED                                                                                                                                                                                                                                                                          [ 66%]
tests/pin_dependency_test.py::test_unpinned_dependencies[tests/resources/tilde.toml] PASSED                                                                                                                                                                                                                                                                          [ 77%]
tests/pin_dependency_test.py::test_unpinned_dependencies[tests/resources/less_than_equal.toml] PASSED                                                                                                                                                                                                                                                                [ 88%]
tests/pin_dependency_test.py::test_unpinned_dependencies[tests/resources/more_than_equal.toml] PASSED                                                                                                                                                                                                                                                                [100%]

================================================================================================================================================================================= FAILURES =================================================================================================================================================================================
__________________________________________________________________________________________________________________________________________________________________________ test_iter_dependencies __________________________________________________________________________________________________________________________________________________________________________

    def test_iter_dependencies() -> None:
        dependencies = iter_dependencies({
            "tool": {
                "poetry": {
                    "dependencies": {
                        "requests": "^2.25.1",
                        "toml": {"version": "^0.10.2"},
                        "torch": [
                            {"version": "^2.2.2"},
                            {"version": "^2.2.2+cpu"},
                        ],
                    }
                }
            }
        })
>       assert list(dependencies) == [
            ("requests", "^2.25.1"),
            ("toml", "^0.10.2"),
        ]
E       AssertionError: assert [('requests', '^2.25.1'), ('toml', '^0.10.2'), ('torch', '^2.2.2'), ('torch', '^2.2.2+cpu')] == [('requests', '^2.25.1'), ('toml', '^0.10.2')]
E         Left contains 2 more items, first extra item: ('torch', '^2.2.2')
E         Full diff:
E           [
E            ('requests',
E             '^2.25.1'),
E            ('toml',
E             '^0.10.2'),
E         +  ('torch',
E         +   '^2.2.2'),
E         +  ('torch',
E         +   '^2.2.2+cpu'),
E           ]

tests/iter_dependencies_test.py:21: AssertionError
```

--- a/tests/iter_dependencies_test.py
+++ b/tests/iter_dependencies_test.py
@@ -21,6 +21,8 @@ def test_iter_dependencies() -> None:
     assert list(dependencies) == [
         ("requests", "^2.25.1"),
         ("toml", "^0.10.2"),
+        ("torch", "^2.2.2"),
+        ("torch", "^2.2.2+cpu"),
     ]
 
 def test_iter_dependencies_no_dependencies() -> None:

